File: 231108/신기한 타일 뒤집기/strange-flipping-tiles.py
tile = [-1] * 200000

n = int(input())
cur = 100000

for _ in range(n):
    num, color = input().split()
    num = int(num)

    if color == "R":
        for i in range(num):
            tile[cur] = 1
            if i != num - 1:
                cur += 1
    else:
        for i in range(num):
            tile[cur] = 0
            if i != num - 1:
                cur -= 1

w, b = 0, 0
for i in tile:
    if i == 0:
        w += 1
    elif i == 1:
        b += 1

print(w, b)

File: 251202/신기한 타일 뒤집기/strange-flipping-tiles.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const commands = input.slice(1);

// Please Write your code here.

const OFFSET = 100000;

const vertical = new Array(OFFSET*2+1).fill(0);
let cur = OFFSET;

for(let command of commands){
    let [distance, direction] = command.split(' ');
    distance = Number(distance);

    if(direction==='R'){
        while(distance>0){
            vertical[cur]=1;
            distance--;
            cur++;
        }
    }else{
        while(distance>0){
            cur--;
            vertical[cur]=2;
            distance--;
        }
    }
}

let b = vertical.filter((e)=>e===1).length;
let w = vertical.filter((e)=>e===2).length;

console.log(w,b)